app/forms/candidate_interface/other_qualification_detail_form.rb

```diff
@@ -0,0 +1,174 @@
+      end
+    end
+
+    def qualification_type_name
+      if qualification_type == OtherQualificationTypeForm::NON_UK_TYPE
+        non_uk_qualification_type
+      elsif qualification_type == OtherQualificationTypeForm::OTHER_TYPE && other_uk_qualification_type.present?
+        other_uk_qualification_type
+      else
+        qualification_type
+      end
+    end
+
+    def title
+      "#{qualification_type_name} #{subject}"
+    end
+
+    PERSISTENT_ATTRIBUTES = %w[qualification_type other_uk_qualification_type non_uk_qualification_type subject predicted_grade grade award_year institution_country].freeze
+    def persistent_attributes(application_qualification)
+      application_qualification.attributes.select { |key, _| PERSISTENT_ATTRIBUTES.include?(key) }
+    end
+
+    def missing_type_validation_error?
+      valid?
+      errors.details[:qualification_type].any? { |e| e[:error] == :blank }
+    end
+
+    def grade_hint
+      if qualification_type == CandidateInterface::OtherQualificationTypeForm::GCSE_TYPE
+        { text: I18n.t('gcse_edit_grade.hint.other.gcse_single_and_double') }
+      end
+    end
+
+    def current_qualification
+      @current_qualification ||= id.present? ? @current_application.application_qualifications.other.find(id) : nil
+    end
+
+  private
+
+    def previous_qualification_is_of_same_type?(qualifications)
+      last_qualification = qualifications[-1]
+      qualification_type == last_qualification.qualification_type
+    end
+
+    def attributes_for_persistence
+      {
+        qualification_type: qualification_type,
+        subject: subject,
+        institution_country: institution_country,
+        predicted_grade: predicted_grade,
+        grade: grade,
+        other_uk_qualification_type: other_uk_qualification_type,
+        non_uk_qualification_type: non_uk_qualification_type,
+        award_year: award_year,
+      }
+    end
+
+    def intermediate_state
+      as_json(
+        only: %w[id current_step checking_answers qualification_type other_uk_qualification_type non_uk_qualification_type subject institution_country predicted_grade grade award_year],
+      )
+    end
+
+    def award_year_is_date_and_before_current_year
+      if !valid_year?(award_year)
+        errors.add(:award_year, :invalid)
+      elsif future_year?(award_year)
+        errors.add(:award_year, :in_the_future)
+      end
+    end
+
+    def grade_format_is_valid
+      case qualification_type
+      when OtherQualificationTypeForm::A_LEVEL_TYPE
+        unless grade.in?(A_LEVEL_GRADES)
+          errors.add(:grade, :invalid)
+        end
+      when OtherQualificationTypeForm::AS_LEVEL_TYPE
+        unless grade.in?(AS_LEVEL_GRADES)
+          errors.add(:grade, :invalid)
+        end
+      end
+    end
+
+    def sanitize_grade_where_required
+      if qualification_type.in? [OtherQualificationTypeForm::A_LEVEL_TYPE, OtherQualificationTypeForm::AS_LEVEL_TYPE]
```

Comment:
We now need GCSE in this list as well, following https://github.com/DFE-Digital/apply-for-teacher-training/pull/3477/files